tests/integration/test_catalog.py

```diff
@@ -0,0 +1,466 @@
+#  with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+from pathlib import Path, PosixPath
+from typing import Generator, List
+
+import pytest
+
+from pyiceberg.catalog import Catalog, MetastoreCatalog
+from pyiceberg.catalog.hive import HiveCatalog
+from pyiceberg.catalog.memory import InMemoryCatalog
+from pyiceberg.catalog.rest import RestCatalog
+from pyiceberg.catalog.sql import SqlCatalog
+from pyiceberg.exceptions import (
+    NamespaceAlreadyExistsError,
+    NamespaceNotEmptyError,
+    NoSuchNamespaceError,
+    NoSuchTableError,
+    TableAlreadyExistsError,
+)
+from pyiceberg.io import WAREHOUSE
+from pyiceberg.schema import Schema
+from tests.conftest import clean_up
+
+
+@pytest.fixture(scope="function")
+def memory_catalog(tmp_path: PosixPath) -> Generator[Catalog, None, None]:
+    test_catalog = InMemoryCatalog(
+        "test.in_memory.catalog", **{WAREHOUSE: tmp_path.absolute().as_posix(), "test.key": "test.value"}
+    )
+    yield test_catalog
+
+    clean_up(test_catalog)
+
+
+@pytest.fixture(scope="function")
+def sqlite_catalog_memory(warehouse: Path) -> Generator[Catalog, None, None]:
+    test_catalog = SqlCatalog("sqlitememory", uri="sqlite:///:memory:", warehouse=f"file://{warehouse}")
+
+    yield test_catalog
+
+    clean_up(test_catalog)
+
+
+@pytest.fixture(scope="function")
+def sqlite_catalog_file(warehouse: Path) -> Generator[Catalog, None, None]:
+    test_catalog = SqlCatalog("sqlitefile", uri=f"sqlite:////{warehouse}/sql-catalog.db", warehouse=f"file://{warehouse}")
+
+    yield test_catalog
+
+    clean_up(test_catalog)
+
+
+@pytest.fixture(scope="function")
+def rest_catalog() -> Generator[Catalog, None, None]:
+    test_catalog = RestCatalog("rest", uri="http://localhost:8181")
+
+    yield test_catalog
+
+    clean_up(test_catalog)
+
+
+@pytest.fixture(scope="function")
+def hive_catalog() -> Generator[Catalog, None, None]:
+    test_catalog = HiveCatalog(
+        "test_hive_catalog",
+        **{
+            "uri": "http://localhost:9083",
+            "s3.endpoint": "http://localhost:9000",
+            "s3.access-key-id": "admin",
+            "s3.secret-access-key": "password",
+        },
+    )
+    yield test_catalog
+    clean_up(test_catalog)
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "test_catalog",
+    [
+        pytest.lazy_fixture("memory_catalog"),
+        pytest.lazy_fixture("sqlite_catalog_memory"),
+        pytest.lazy_fixture("sqlite_catalog_file"),
+        pytest.lazy_fixture("rest_catalog"),
+        pytest.lazy_fixture("hive_catalog"),
+    ],
+)
```

Comment:
How about moving this to a common variable instead of repeating this for each of the tests?

```suggestion
CATALOGS = [
    pytest.lazy_fixture("memory_catalog"),
    pytest.lazy_fixture("sqlite_catalog_memory"),
    pytest.lazy_fixture("sqlite_catalog_file"),
    pytest.lazy_fixture("rest_catalog"),
    pytest.lazy_fixture("hive_catalog"),
]

@pytest.mark.integration
@pytest.mark.parametrize("test_catalog", CATALOGS)
```

Reply:
nice, added